core/DataTable/Filter/ReplaceSummaryRowLabel.php

```diff
@@ -55,11 +55,6 @@ public function filter($table)
         $row = $table->getRowFromId(DataTable::ID_SUMMARY_ROW);
         if ($row) {
             $row->setColumn('label', $this->newLabel);
-        } else {
-            $row = $table->getRowFromLabel(DataTable::LABEL_SUMMARY_ROW);
```

Comment:
not sure, just a random thought... would we need to iterate over all rows and call `isSummaryRow()` to find a summary row? Not sure when it would have gone into this code path. Or actually we could maybe expose a method 

```php
function getSummaryRow() { return $this->summaryRow; }
```

to the datatable additionally? Although probably not needed if dataTable makes sure it always has the right index

Reply:
No, we can do `getRowFromId(DataTable::ID_SUMMARY_ROW);`